src/core/room-lifecycle-manager.ts

```diff
@@ -1,923 +1,394 @@
-   */
-  private readonly _mtx = new Mutex();
-
-  /**
-   * A map of contributors to transient detach timeouts.
-   *
-   * If a channel enters the attaching state (as a result of a server initiated detach), we should initially
-   * consider it to be transient and not bother changing the room status.
-   */
-  private readonly _transientDetachTimeouts: Map<ContributesToRoomLifecycle, ReturnType<typeof setTimeout>>;
-
-  /**
-   * This flag indicates whether some sort of controlled operation is in progress (e.g. attaching, detaching, releasing).
-   *
-   * It is used to prevent the room status from being changed by individual channel state changes and ignore
-   * underlying channel events until we reach a consistent state.
-   */
-  private _operationInProgress = false;
-
-  /**
-   * A map of pending discontinuity events.
-   *
-   * When a discontinuity happens due to a failed resume, we don't want to surface that until the room is consistently
-   * attached again. This map allows us to queue up discontinuity events until we're ready to process them.
-   */
-  private _pendingDiscontinuityEvents: DiscontinuityEventMap = new Map();
-
-  /**
-   * A map of contributors to whether their first attach has completed.
-   *
-   * Used to control whether we should trigger discontinuity events.
-   */
-  private _firstAttachesCompleted = new Map<ContributesToRoomLifecycle, boolean>();
-
-  /**
-   * Are we in the process of releasing the room?
-   */
-  private _releaseInProgress = false;
-
-  /**
-   * Constructs a new `RoomLifecycleManager` instance.
-   * @param lifecycle The room lifecycle that manages status.
-   * @param contributors The features that contribute to the room status.
-   * @param logger An instance of the Logger.
-   * @param transientDetachTimeout The number of milliseconds to consider a detach to be "transient"
-   */
-  constructor(
-    lifecycle: DefaultRoomLifecycle,
-    contributors: ContributesToRoomLifecycle[],
-    logger: Logger,
-    transientDetachTimeout: number,
-  ) {
+  private readonly _roomId: string;
+  private readonly _eventEmitter: EventEmitter<RoomLifeCycleEvents>;
+  private _isFirstAttach: boolean; // CHA-RL13
+  private _isPostExplicitDetach: boolean; // CHA-RL14
+  private readonly _mutex: Mutex; // CHA-RL7
+  private _operationInProgress: boolean;
+
+  constructor(roomId: string, channelManager: ChannelManager, roomLifecycle: InternalRoomLifecycle, logger: Logger) {
+    this._roomId = roomId;
+    this._channelManager = channelManager;
+    this._roomLifecycle = roomLifecycle;
     this._logger = logger;
-    this._contributors = contributors;
-    this._transientDetachTimeouts = new Map();
-    this._lifecycle = lifecycle;
+    this._eventEmitter = new EventEmitter();
+    this._isFirstAttach = true; // CHA-RL13
+    this._isPostExplicitDetach = false; // CHA-RL14
+    this._mutex = new Mutex(); // CHA-RL7
+    this._operationInProgress = false;
```

Comment:
```suggestion
    this._mutex = new Mutex(); // CHA-RL7
    // this._operationInProgress not needed
```


`_operationInProgress` indicates whether `this._mutex` has an operation in progress.
I firmly believe check for the same can be replaced by `this._mutex.isLocked()`, [doc link](https://github.com/DirtyHairy/async-mutex/tree/master?tab=readme-ov-file#checking-whether-the-mutex-is-locked)
Since JS is single-threaded, it should result in exact same behaviour.